tools/convert_feat_to_wholegraph.py

```diff
@@ -0,0 +1,104 @@
+"""
+    Copyright 2023 Contributors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Tool to convert node features from distDGL format to WholeGraph format.
+"""
+import argparse
+import json
+import os
+import re
+import torch
+import dgl
+import pylibwholegraph.torch as wgth
+
+
+def main(folder):
+    """ Convert node features from distDGL format to WholeGraph format"""
+    os.environ['MASTER_ADDR'] = '127.0.0.1'
+    os.environ['MASTER_PORT'] = '1234'
+    torch.distributed.init_process_group('gloo', world_size=1, rank=0)
+    local_comm = wgth.comm.get_local_device_communicator()
+    wg_folder = os.path.join(folder, 'wholegraph')
+    if not os.path.exists(wg_folder):
+        os.makedirs(wg_folder)
+
+    # TODO(IN): Add support for edge features
+    node_feats_data = []
+    folder_pattern = re.compile(r"^part[0-9]+$")
+    for path in (os.path.join(folder, name) for name in sorted( \
+        (f for f in os.listdir(folder) if os.path.isdir(os.path.join(folder, f)) \
+        and folder_pattern.match(f)), key=lambda x: int(x.split("part")[1]))):
+        node_feats_data.append(dgl.data.utils.load_tensors(f'{path}/node_feat.dgl'))
+
+    num_parts = len(node_feats_data)
+    metadata = {}
+    masks = ["train_mask", "test_mask", "val_mask", "labels"]
+
+    for feat in list(node_feats_data[0].keys()):
+        if feat.split('/')[1] not in masks:
+            print(f"Processing '{feat}' features...")
+            if feat not in node_feats_data[0]:
```

Comment:
You get ```feat``` from ```list(node_feats_data[0].keys())```, Why this if statement can be true?

Reply:
Right! It doesn't make sense anymore.